### DISCUSSION TITLE
Calling multiple dataloaders in a row

### DISCUSSION THREAD

**Discussion by kutlueren** [Netflix/dgs-framework#1333] [Q&A]
Hi 

We are using the library in our Graphql server written in Kotlin. I have a query very similar to the one [here](https://github.com/Netflix/dgs-framework/discussions/872) (and I tried the solution proposed to solve the issue we have but it did not) :

```
type Query {
    getPropertyDetails(
        input: PropertyDetailsInput!
    ): PropertyDetailsResponse
}

input PropertyDetailsInput {
    id: [ID!]!
}

type propertyDetailsResponse {
    propertyDetails: [propertyDetails!]
}

type propertyDetails {
    id: ID!
    ... """Bunch of other fields"""
    details: Details
}

type Details {
    address: String!
    ... """Bunch of other fields"""
}
```

So when we run the query below:

```
query getProperties(
  $input: PropertyDetailsInput!
) {
  getPropertyDetails(input: $input) {
    propertyDetails {
      details{
        address
        ... """Bunch of other fields"""
      }
    }
  }
}
```

The funtion which resolves the query returns all the properties given in `PropertyDetailsInput`. We are using a `dataloader` to fetch the properties. Let's call it `dataloader-1`. Then the funtion to resolve the detail field is being called and then we are using another `dataloader` to fetch the details. Let's call that one `dataloader-2`. In this case `dataloader-2` is not responding and just hanging with no result. If I omit `details` from the query then it runs perfectly. 

```
 val dataLoader-1: DataLoader<Id, PropertyDetails?> =
            dfe.getDataLoader(PropertyDataLoader::class.java)
return dataLoader-1.load(ids).await()
            
val dataLoader-2: DataLoader<Id, DetailsDataLoaderResult?> =
                dfe.getDataLoader(DetailsDataLoader::class.java)
return dataLoader-2.load(ids).await()
```

Can anyone help me navigate that? Has anyone encountered the same issue before?

**Comment by bjoernmayer** [comment]
This is resolved.
Instead of returning `CompletableFutures` inside the data loader methods, we used Kotlin's `await`.
Then DGS didn't trigger the Futures to do their work and hence the `await` waited forever
